bucket_test.go

```diff
@@ -908,6 +908,49 @@ func TestBucket(t *testing.T) {
 						}
 					}
 				})
+
+				if !strings.Contains(impl.name, "S3BucketChecksums") {
+					return
+				}
+
+				t.Run("PullWithCache", func(t *testing.T) {
```

Comment:
I would move this to `s3_bucket_util_test.go` since it is specific to S3 buckets.

Reply:
It is specific to S3 buckets but large vs. small isn't relevant - how would you suggest I add that? Just have both the large and small GetS3BucketTests function add the same test to each? It looked like there was already precedent for excluding certain implementations from bucket_test for the case of GridFS.

Reply:
I think it will fail for large buckets, no? Since the ETag isn't the md5 hash of the file? If I am correct in my understanding, you can just add it for the small bucket tests until we figure out what to do for checksums against multi-part uploaded files.

Reply:
Sorry deleted the last comment, I misspoke. Yes, I think I want to refactor these tests but for now duplicating the tests for both small and large is fine.

Reply:
I actually misunderstood the difference between large and small buckets! Large uses multi-part uploads, which is very much relevant given that we know our use of ETag won't work for those. I am going to avoid testing this for large for now, and then when we have a more robust comparison we can add the same test to large.